### PR TITLE
Record source leakage validation summaries

### LINKED ISSUE #740: OpenSearch-VL alignment: add data construction and leakage controls - Milestone 2: add leakage and overlap validators - Unit 2: Record leakage validation summaries in dataset manifests and evaluation artifacts.
## Executable Unit

Record leakage validation summaries in dataset manifests and evaluation artifacts.

## Parent Context

- Direction: `OpenSearch-VL alignment: add data construction and leakage controls`
- Plan/Milestone: `Milestone 2: add leakage and overlap validators`

## Definition Of Done

- The change is scoped to the smallest useful implementation slice.
- The governing spec or plan is created or updated before broad code changes.
- Tests cover the new behavior or contract edge case.
- Metrics and verification output are recorded in the PR evidence.

## Suggested Verification

- `git diff --check`
- Focused pytest or Swift test for the touched code path
- Changed-scope coverage for modified Python or Swift files when measurable
- Any relevant benchmark/evaluation artifact if this changes measured behavior

### PR DESCRIPTION
**Pull request by Zigfreidish**
## Summary
- Record publishable `source_leakage_validation` summaries in source-anchored training and final-result evaluation package manifests.
- Capture validator names, row/segment counts, split source-id counts, and the lexical threshold without persisting raw excluded values or prompt/observation text.
- Update the source-anchored construction plan and benchmark/evaluation contract with the artifact summary contract.

Closes #740.

## Plan or Spec
- `docs/plans/2026-05-22-source-anchored-data-construction.md`
- `docs/benchmark-evaluation-contract.md`

## Commands Run
```text
python3 -m py_compile services/mlx-worker-python/worker/productization/synthetic_dataset_generation.py services/mlx-worker-python/tests/test_synthetic_dataset_generation.py
# passed

git diff --check
# passed

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_synthetic_dataset_generation.py -k "source_leakage_validation or source_construction"
# 23 passed, 47 deselected

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python pytest -q services/mlx-worker-python/tests/test_synthetic_dataset_generation.py
# 70 passed

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage run --data-file=/tmp/leakage_summaries.coverage -m pytest -q services/mlx-worker-python/tests/test_synthetic_dataset_generation.py
# 70 passed

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python coverage json --data-file=/tmp/leakage_summaries.coverage -o /tmp/leakage_summaries_coverage.json
# wrote /tmp/leakage_summaries_coverage.json

PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/changed_scope_coverage.py --coverage-json /tmp/leakage_summaries_coverage.json services/mlx-worker-python/worker/productization/synthetic_dataset_generation.py
# TOTAL 41 0 100%

git diff --name-only origin/main | jq -R -s 'split("\n")[:-1]' > /tmp/leakage_summaries_changed_files.json
PYTHONPATH="$PWD:$PWD/services/mlx-worker-python" uv run --frozen --project services/mlx-worker-python python scripts/pr_scoped_performance_scope.py --registry infra/perf/pr_scoped_probes.json --changed-files-json /tmp/leakage_summaries_changed_files.json --output /tmp/leakage_summaries_scope.json
# selected_count: 0
```

## Coverage and Metrics
- Changed-scope coverage: `TOTAL 41 0 100%` for `synthetic_dataset_generation.py`.
- Metrics report: local PR-scoped performance scope selected `0` probes. This change persists package-manifest evidence for already-running validation logic; it does not change runtime request paths, model execution, or benchmark/evaluation scoring loops.
- Observability mode: `evidence`; manifest summaries make leakage validation outcomes auditable in release-facing dataset artifacts.
- Probe overhead: `N/A`; no runtime probes, counters, or debug instrumentation were added.

## Known Gaps
- Full local `make bootstrap`, `make proto`, `make swift-test`, `make py-test`, and `make integration-test` were not run on this host because the filesystem has about 3.0 GiB free.
- No benchmark/evaluation runtime artifact was generated because this change is limited to synthetic package manifest materialization.

## Evidence Checklist
- [x] Relevant plan or spec is identified.
- [x] Behavior changes are reflected in the relevant docs.
- [x] Protocol changes include regenerated generated artifacts.
- [x] Dependency changes include updated lockfiles.
- [x] Relevant tests were run.
- [x] A metrics report is included, or `N/A` is stated explicitly with the reason.
- [x] Observability mode, probe overhead, and any deferred debug or evidence work are recorded, or `N/A` is stated explicitly with the reason.
- [x] Deferred work and known gaps are stated explicitly.

### CODE REVIEW

File: services/mlx-worker-python/worker/productization/synthetic_dataset_generation.py [hunk 1]
```diff
@@ -1193,12 +1257,24 @@ def _first_leakage_match(
...
+        summary.shared_source_id_count = 0
     if not train_rows or not validation_rows:
         return
     train_source_ids = _source_ids_by_split(train_rows)
     validation_source_ids = _source_ids_by_split(validation_rows)
     shared_ids = sorted(set(train_source_ids) & set(validation_source_ids))
+    if summary is not None:
+        summary.train_source_id_count = len(train_source_ids)
+        summary.validation_source_id_count = len(validation_source_ids)
+        summary.shared_source_id_count = len(shared_ids)
```
[the remark this answers is not in the document]
Comment (Zigfreidish) [reply services/mlx-worker-python/worker/productization/synthetic_dataset_generation.py:1277]: Fixed in 8ef5a8c1 by computing train and validation source-id maps once in _validate_source_split_collision and reusing them for both summary counts and collision checks. Re-ran focused/full synthetic dataset tests and changed-scope coverage.
